### ISSUE TITLE
binding after a click event triggers the onEvent function 

### ISSUE THREAD

**Issue by mjntan35** [ndelvalle/v-click-outside#41]
The problem is when the bind function is called after a click event, the event handler that is added to the document's event listener will also be triggered

https://jsfiddle.net/yjxgawpk/
this fiddle illustrate that adding an eventlistener to document after an click event. 
Notice how without using setTimeout, the evenhandler is triggered after adding it

**Comment by Skasi** [comment]
I've got the same issue. It first came up with Nuxt child elements only some months ago, but now after some more updates it happens for any element that works the way described above. I assume it's due to a Vue change or something along those lines.

**Comment by jericopulvera** [comment]
Hello. @Skasi @mjntan35

I'm facing the exact issue.
Just wondering if you've found a workaround for this?

**Comment by jericopulvera** [comment]
used https://github.com/simplesmiler/vue-clickaway instead and all is working now.

**Comment by ndelvalle** [comment]
@Skasi good reference 👍 do you mind open a PR to fix this, please.

**Comment by chasegiunta** [comment]
Any update on this? Every site where I used v-click-outside to create modals is broke, ha 😅

**Comment by ndelvalle** [comment]
@chasegiunta I haven't had time to work on this yet, probably this weekend, if someone takes a look at it and create a PR it would be great 👍

**Comment by mjntan35** [comment]
@ndelvalle I debugged the problem, it seems to work if you use setTimeout, it works

setTimeout(document.addEventListener,0, event, handler)

https://github.com/ndelvalle/v-click-outside/pull/42

**Comment by monkeyWzr** [comment]
@mjntan35 
Thank you! But I think the update function should also be fixed.
[L102](https://github.com/ndelvalle/v-click-outside/blob/832802fa5408bea162195adf27a1dbb262d5361c/lib/v-click-outside.js#L102)

**Comment by mjntan35** [comment]
@monkeyWzr thanks for the comment.. I have modified the PR

**Comment by ndelvalle** [comment]
Published under v-click-outside@2.1.0-0 or using `next` tag. I need to update some test from changes merged to master in a previous PR before creating a release.

**Comment by ndelvalle** [comment]
Deployed `v-click-outside@2.1.0-1`

**Comment by zanedev** [comment]
@ndelvalle just a heads up I tried out the `next` release and found a few issues so I downgraded back to latest 2.0 release and it fixed it for me.  I don't have time to repro it for you but this is what I discovered in `next`:

- The click outside event was firing even though the component it was used on with a directive was destroyed using a v-if.  It has a top level vue transition in the template if that has anything to do with it..
- The second time it mounted I was getting click outside events firing on children inside the component.  The first time it renders it did not do this.

Also I noticed in both versions that dynamically loaded/async children components that render after mount were not captured as inside clicks, they are reporting an outside click even though they are children.
